f5_openstack_agent/lbaasv2/drivers/bigip/agent_manager.py

```diff
@@ -627,8 +642,13 @@ def remove_orphans(self, all_loadbalancers):
     def create_loadbalancer(self, context, loadbalancer, service):
         """Handle RPC cast from plugin to create_loadbalancer."""
         try:
-            self.lbdriver.create_loadbalancer(loadbalancer, service)
+            service_pending = \
+                self.lbdriver.create_loadbalancer(loadbalancer,
+                                                  service)
             self.cache.put(service, self.agent_host)
+            if service_pending:
+                self.needs_resync = True
+
```

Comment:
If two different lb create methods are called, and the first one fails, but the second succeeds the second success will clear away the needs_resync.  Possibly this could occur before we can call the refresh on the pending service.